File: typescript-sdk/src/sdk/models/shared/updateroom.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class UpdateRoom extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=defaultAccesses" })
  defaultAccesses?: any;

  @SpeakeasyMetadata({ data: "json, name=groupsAccesses" })
  groupsAccesses?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=usersAccesses" })
  usersAccesses?: Record<string, any>;
}

File: go-sdk/pkg/models/shared/updateroom.go
package shared

type UpdateRoom struct {
	DefaultAccesses *interface{}           `json:"defaultAccesses,omitempty"`
	GroupsAccesses  map[string]interface{} `json:"groupsAccesses,omitempty"`
	Metadata        map[string]interface{} `json:"metadata,omitempty"`
	UsersAccesses   map[string]interface{} `json:"usersAccesses,omitempty"`
}
